src/go/k8s/api/redpanda/v1alpha1/redpanda_conversion.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+package v1alpha1
+
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	v1alpha2 "github.com/redpanda-data/redpanda-operator/src/go/k8s/api/redpanda/v1alpha2"
+)
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (r *Redpanda) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// ConvertTo converts Redpanda to the Hub version (v1alpha2).
+func (src *Redpanda) ConvertTo(dstRaw conversion.Hub) error {
```

Comment:
Could we add some unit tests for this and ConvertFrom? Randomized testing would be a good fit here IMO so we can assert that all fields are getting translated as expected.

Reply:
Great suggestion I will create it.